### PR TITLE
Update trait to be as specified in OAM spec

### PR DESCRIPTION
**Pull request by iyyappam**
Addressing issue https://github.com/oam-dev/rudr/issues/478 

Updating the traits input to adhere to the strict schema as specified in OAM spec.

Adding version info in the traits name, so that it enables updating/rolling different version of the trait. (We can discuss introducing a new field named "type" which might be the right way to do it).

For backward compatibility, I have kept the exiting trait name as it is. But after the next milestone, we could remove them.

### CODE REVIEW

File: src/trait_manager.rs [hunk 1]
```diff
@@ -61,7 +111,7 @@ impl TraitManager {
                 );
                 Ok(OAMTrait::Ingress(ing))
             }
-            traits::VOLUME_MOUNTER => {
+           traits::VOLUME_MOUNTER => {
```
Comment (hongchaodeng) [inline src/trait_manager.rs]: indent not correct?

**Comment by wonderflow** [comment]
Cool! You could run `cargo clippy` to fix the CI

File: src/trait_manager.rs (hunk 1 above)
Comment (iyyappam) [reply src/trait_manager.rs]: I started with enum as it is recommended way in Rust but I found it restrictive. For example, named struct cannot be added without changing the schema or if anyone wants to add a new trait, they didn't have to update that enum. So I decided on parsing as JSON value which could be interpreted by the trait implementer.

I'll fix the indentation bug.

**Comment by iyyappam** [comment]
Sure. I'll update after clippy run.

**Comment by wonderflow** [comment]
This is really good start! I think we could do same thing on workloadType

**Comment by hongchaodeng** [comment]
lgtm

**Pull request merged**